engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/transform/BpmnResourceTransformer.java

```diff
@@ -69,9 +70,13 @@ public Either<Failure, Void> transformResource(
               if (validationError == null) {
                 // transform the model to avoid unexpected failures that are not covered by the
                 // validator
-                bpmnTransformer.transformDefinitions(definition);
+                final var executableProcesses = bpmnTransformer.transformDefinitions(definition);
 
                 return checkForDuplicateBpmnId(definition, resource, deployment)
+                    .flatMap(
+                        unused ->
+                            StraightThroughProcessingLoopValidator.validate(
```

Comment:
❓ wdyt about adding an experimental flag to be able to disable it at will (in case of false-positives)

Reply:
I also would up for a feature-flag

Reply:
I agree, I will add it with the other FeatureFlags though if that's okay. So instead of `broker.engine.validation.noWaitStateLoopRejection: true/false` it would be `broker.experimental.features.enableStraightThroughProcessingLoopDetector: true/false`